keel-core-test/src/main/kotlin/com/netflix/spinnaker/keel/persistence/VerificationRepositoryTests.kt

```diff
@@ -218,44 +145,54 @@ abstract class VerificationRepositoryTests<IMPLEMENTATION : VerificationReposito
       .with(VerificationState::metadata) { isEqualTo(initialMetadata + newMetadata) }
   }
 
+  @Test
+  fun `successive updates can append to arrays in the metadata`() {
+    context.setup()
+
+    subject.updateState(context, verification, PENDING)
+    val initialMetadata = mapOf("tasks" to listOf(ULID().nextULID()))
+    subject.updateState(context, verification, FAIL, initialMetadata)
+    val newMetadata = mapOf("tasks" to listOf(ULID().nextULID()))
+    subject.updateState(context, verification, PASS, newMetadata)
+
+    expectCatching {
+      subject.getState(context, verification)
+    }
+      .isSuccess()
+      .isNotNull()
+      .with(VerificationState::metadata) {
+        get("tasks").isA<List<*>>().hasSize(2)
+      }
+  }
```

Comment:
This tests that if you update with metadata containing an array property it merges with any existing array data